voxblox/include/voxblox/integrator/tsdf_integrator.h

```diff
@@ -155,109 +316,93 @@ class TsdfIntegratorBase {
   FloatingPoint voxel_size_inv_;
   FloatingPoint voxels_per_side_inv_;
   FloatingPoint block_size_inv_;
+
+  std::shared_ptr<PointGetter> point_getter_;
+  ApproxHashArray<12, std::mutex> mutexes_;  // 4096 locks
 };
 
 class SimpleTsdfIntegrator : public TsdfIntegratorBase {
  public:
   SimpleTsdfIntegrator(const Config& config, Layer<TsdfVoxel>* layer)
-      : TsdfIntegratorBase(config, layer){};
+      : TsdfIntegratorBase(config, layer) {}
 
   void integratePointCloud(const Transformation& T_G_C,
                            const Pointcloud& points_C, const Colors& colors) {
-    DCHECK_EQ(points_C.size(), colors.size());
-    timing::Timer integrate_timer("integrate");
-
-    const Point& origin = T_G_C.getPosition();
-
-    for (size_t pt_idx = 0; pt_idx < points_C.size(); ++pt_idx) {
-      const Point& point_C = points_C[pt_idx];
-      const Point point_G = T_G_C * point_C;
-      const Color& color = colors[pt_idx];
-
-      FloatingPoint ray_distance = (point_G - origin).norm();
-      if (ray_distance < config_.min_ray_length_m) {
-        continue;
-      } else if (ray_distance > config_.max_ray_length_m) {
-        // TODO(helenol): clear until max ray length instead.
-        continue;
-      }
+    timing::Timer block_timer("block_allocation");
+    allocateBlocks(T_G_C, points_C, colors);
+    block_timer.Stop();
 
-      FloatingPoint truncation_distance = config_.default_truncation_distance;
+    timing::Timer integrate_timer("integrate");
 
-      const Ray unit_ray = (point_G - origin).normalized();
+    point_getter_ =
```

Comment:
Because it needs to know T_G_C, points, etc. on construction and is constructed new every integration request, not once per integrator object?